velox/functions/lib/Re2Functions.cpp

```diff
@@ -376,9 +377,32 @@ class Re2SearchAndExtract final : public VectorFunction {
 
 class LikeConstantPattern final : public VectorFunction {
  public:
+  // Constructor to preprocess and save the state of the string 'pattern' to be
+  // matched in 'like' expression, specifically for the following cases: 1)
+  // Constant pattern search: 'pattern' is an exact pattern i.e it has no
+  // wildcard ('_', '%') or escape characters. 2) 'pattern' has an exact pattern
+  // as prefix followed by one or more '%'. 3) 'pattern' has an exact pattern as
+  // suffix preceded by one or more '%'. If the pattern matches any of
+  // these three cases, we maintain the state, which allows us to invoke a
+  // simplified pattern match function.
   LikeConstantPattern(StringView pattern, std::optional<char> escapeChar)
       : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
-            RE2::Quiet) {}
+            RE2::Quiet) {
+    pattern_ = pattern;
+    hasConstPattern_ = false;
+    hasPrefixPattern_ = false;
+    hasSuffixPattern_ = false;
+
+    if (escapeChar) {
+      return;
+    } else if (!likeutils::checkWildcardInPattern(pattern_)) {
+      hasConstPattern_ = true;
+    } else if (likeutils::isPrefixPattern(pattern_)) {
```

Comment:
Rename variables to fixedPattern_, prefixPattern_ and suffixPattern_ (remove the has prefix)